### ISSUE TITLE
2D RCS gives wrong results

### ISSUE THREAD

**Issue by Peter1511ogluigi** [NanoComp/meep#1933]
Hello

I am trying to compute the RCS of an object by following the tutorial for differential-cross section (https://meep.readthedocs.io/en/latest/Python_Tutorials/Basics/#differentialradar-cross-section). 
I am only interested in a 2D scenario, so I have tried to change the code such that the problem is reduced to 2D. Since I am interested in computing the RCS, I am only looking at the backscattered waves, therefore I have a setup that looks like the following 

![setup_RCS](https://user-images.githubusercontent.com/47326008/153190073-15301955-9bdf-4b0d-83d8-4a8549b7a99a.png)

Here the near field box is reduced to a line, shown in blue and the source is shown as a red line while the white square is a metal block.

When I compute the RCS vs far-field point I get that the RCS grows with distance which seems wrong  as described in https://github.com/NanoComp/meep/issues/1055#issuecomment-554714522 . 

![distanceloglog](https://user-images.githubusercontent.com/47326008/153193859-ff768092-76b7-4cef-b640-d997ca8e7673.png)

I don't know why my RCS grows with distance (or why it is as large as it is). I suspect that I have misunderstood something in the documentation behind calculation of RCS with the near to far field method. 
If someone can spot my mistake it would really save my day

/Peter

Here is the code:
`
    import meep as mp
    import numpy as np
    import matplotlib.pyplot as plt
    import PyMieScatt as ps
    import math

    angle_points =  1    #define how many sections my rotation is divided into

    frq_cen = 1.0
    resolution = 40 # pixels/um
    dpml = 0.5
    dair = 1.5 # at least 0.5/frq_cen padding between source and near-field monitor

    pml_layers = [mp.PML(thickness=dpml)]
    s = 2*(dpml+dair+1)
    cell_size = mp.Vector3(s,s)
    #---------------------------------- Define Chirped pulse as the source ------------------------------------------------

    a = 0.2   						# Gaussian envelope half-width
    b = 0  							# linear chirp rate (positive: up-chirp, negative: down-chirp)
    t0 = 0   						# peak time

    chirp = lambda t: np.exp(1j*2*np.pi*frq_cen*(t-t0)) * np.exp(-a*(t-t0)**2+1j*b*(t-t0)**2) #vi gør brug af chirp puls, da dette er mest repræsentativt af virkeligheden

    sources = [mp.Source(src=mp.CustomSource(src_func=chirp,is_integrated=True),
                     center=mp.Vector3(-0.5*s+dpml,0),
			size=mp.Vector3(0,s),
                     component=mp.Ez)]

    #--------------------------------setup before scattering-----------------------------------------------------------------------

    sim = mp.Simulation(resolution=resolution,
                    cell_size=cell_size,
                    boundary_layers=pml_layers,
                    sources=sources,
                    k_point=mp.Vector3())

    box_flux = sim.add_flux(frq_cen, 0, 1,
                        mp.FluxRegion(center=mp.Vector3(x=-1),size=mp.Vector3(0,1)))

    nearfield_box = sim.add_near2far(frq_cen, 0, 1,
                                 mp.Near2FarRegion(center=mp.Vector3(-1,0,0),size=mp.Vector3(0,1),weight=+1)) #choose wheight +1 since we         are interested in incoming flux
                                

    sim.run(until=50)

    input_flux = mp.get_fluxes(box_flux)[0]
    nearfield_box_data = sim.get_near2far_data(nearfield_box)

    sim.reset_meep()

    #-------------------------------setup with scattering ---------------------------------------------------------------------------

    rot_angle = np.pi/(angle_points)   #define angle which object os rotated with in [0,pi]

    #metallic block
    geometry = [mp.Block(center=mp.Vector3(0,0,0),
                     size=mp.Vector3(1,1,0),
                     e1=mp.Vector3(x=1).rotate(mp.Vector3(z=1), rot_angle),
                     e2=mp.Vector3(y=1).rotate(mp.Vector3(z=1), rot_angle),
                     material=mp.metal)]

    sim = mp.Simulation(resolution=resolution,
                    cell_size=cell_size,
                    boundary_layers=pml_layers,
                    sources=sources,
                    k_point=mp.Vector3(),
                    geometry=geometry)

    nearfield_box = sim.add_near2far(frq_cen, 0, 1,
                                 mp.Near2FarRegion(center=mp.Vector3(-1,0,0),size=mp.Vector3(0,1),weight=-1)) # choose wheight -1 since we     now are interested in outgoing flux

    sim.load_minus_near2far_data(nearfield_box, nearfield_box_data)

    sim.run(mp.at_beginning(mp.output_epsilon),mp.to_appended("ez", mp.at_every(0.6, mp.output_efield_z)),until=50)
    npts = 1   #we are only interested in the flux for backscattering

        
    ff_r = np.logspace(2,5,10) # radius of far-field semi-circle

    E = np.zeros((npts,3),dtype=np.complex128)
    H = np.zeros((npts,3),dtype=np.complex128)
    for n in range(npts):

        ff = sim.get_farfield(nearfield_box, mp.Vector3(-ff_r[n]))        
        E[n,:] = [np.conj(ff[j]) for j in range(3)]
        H[n,:] = [ff[j+3] for j in range(3)]

    # compute Poynting flux Pr and RCS

    Px = np.real(np.multiply(E[:,1],H[:,2])-np.multiply(E[:,2],H[:,1]))

    intensity = input_flux/(4*1)**2 #examples I have seen are with a sphere of r=1, so i just wrote 1

    Radar_cross_section = 4 * np.pi * ff_r**2 * Px / intensity

    print("The ff_r is given as")
    print(ff_r)
    print()
    print("The RCS is given as")
    print(Radar_cross_section)

    plt.figure(1)
    plt.loglog(ff_r,Radar_cross_section,'-r',markersize=10)
    plt.xlabel('ff_r')
    plt.ylabel('Radar Cross Section')
    plt.show()

`

**Comment by oskooi** [comment]
> Here the near field box is reduced to a line

This is incorrect. You need to surround the scatterer with a *closed* box exactly as demonstrated in the tutorial.
